website/src/_posts/2022-09-3.0.md

```diff
@@ -0,0 +1,158 @@
+
+But, it didn’t come without its own set of new problems:
+
+* It tries to do the exact same, but it looks like a different product.
+* It’s confusing for users whether they want to use Robodog or Uppy directly.
+* Robodog is more ergonomic because it’s limited. When you hit such a limit, you need to refactor everything to Uppy with plugins.
+
+This has now led us to deprecating Robodog and embrace Uppy for its strong suits: modularity and flexibility. At the same time, we also introduced something to take away some repetitive integration code: [`@uppy/remote-sources`](/docs/remote-sources).
+
+Are you using Robodog in your code base? You can check out the [migration guide](https://uppy.io/docs/migration-guides.html#Migrate-from-Robodog-to-Uppy-plugins) to see how you can transition to Uppy plugins without losing functionality. We have committed ourselves to extend Robodog support for one year to give our users more time to migrate.
+
+## Native mobile camera
+
+As you know, Uppy’s Webcam plugin provides a nice UI for desktop devices to take pictures and record videos. Up until 3.0, the same UI was utilized on mobile as well.
+
+But most mobile devices have system UI for taking pictures and recording videos, which is better than any custom solution we can offer. So we’ve introduced a new option `mobileNativeCamera` for the Webcam plugin:
+
+```js
+uppy.use(Webcam, {
+  mobileNativeCamera: isMobile({ tablet: true }),
+})
+```
+
+By default it uses the [`is-mobile`](https://github.com/juliangruber/is-mobile) package to detect mobile devices, like smartphones and tablets, but you can manually set this option to `true` or `false` if you wish. When enabled, instead of the usual Camera/Webcam Uppy UI, users will be presented with buttons that open native camera interface:
+
+<img alt="Uppy native camera UI" src="/images/blog/3.0/native-camera.jpg">
+
+[`videoConstraints.facingMode`](/docs/webcam/#videoConstraints) is also supported by this option via the [`capture`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/capture) attribute (in the browsers and devices that respect it).
+
+```js
+uppy.use(Webcam, {
+  videoConstraints: {
+    facingMode: 'user',
+  },
+})
+```
+
+Same applies to [`modes`](/docs/webcam/#modes), so you can enable only video or only photos.
+
+## Devotedly stable
+
+Since the Uppy 2.7.0 release, **we’ve fixed over 95 bugs!** This includes bugs that made uploading less stable, documentation corrections, and dependency upgrades.
+
+## Companion turns 4.0
+
+Everyone’s favorite Companion also received some love.
+
+### Streaming upload
+
+Streaming upload can now also be enabled in Companion when using Tus. This comes with greatly improved upload speeds and allows uploading potentially hundreds of gigabytes without needing a large server storage. This feature was also available before Companion 4, but it didn’t work with Tus yet.
```

Comment:
```suggestion
Streaming upload can now also be enabled in Companion when using Tus. This comes with greatly improved upload speeds and allows uploading potentially hundreds of gigabytes without needing a large server storage. [We found that](https://github.com/transloadit/uppy/pull/4046#issuecomment-1235697937) this improves speeds by about 37% for a Google Drive upload of a 1GB file. This feature was also available before Companion 4, but it didn’t work with Tus until now.
```

Reply:
-1 on using `We found that` as the text for the link, I'd much rather prefer to have `(source)` at the end of the sentence.